lib/teach-api.js

```diff
@@ -53,37 +60,85 @@ _.extend(TeachAPI.prototype, {
     var info = this.getLoginInfo();
     return info && info.username;
   },
+
+  // The first half of the oauth2 login work flow:
+  //
+  // Form an oauth2 URL that users can be redirected to, which will eventually
+  // lead to continueLogin being called when the oauth2 service is done handling
+  // the authentication and remote login of the user.
   startLogin: function() {
-    if (!(process.browser && window.navigator.id)) {
-      return this.emit('login:error',
-                       new Error('navigator.id does not exist'));
-    }
-    window.navigator.id.get(function(assertion) {
-      if (!assertion) {
-        return this.emit('login:cancel');
-      }
+    var state = generateRandom();
+    window.sessionStorage['oauth2_token'] = state;
+    var queryArguments = {
+      client_id: "test",
+      response_type: "code",
+      scopes: "user",
+      state: state
+    };
+    query = Object.keys(queryArguments).map(function(key) {
```

Comment:
you can use [url.format()](https://nodejs.org/api/url.html#url_url_format_urlobj) for this